### PR TITLE
Fix for elasticsearch.index_template_present fails on ES <6.x #48442

### PR DESCRIPTION
**Pull request by babs**
### What does this PR do?

It resolv a bug related to index template for Elasticsearch before 6.x

### What issues does this PR fix or reference?

It fixes issue #48442.

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/states/elasticsearch.py [hunk 1]
```diff
@@ -275,10 +275,16 @@ def index_template_present(name, definition, check_definition=False):
                     ret['comment'] = 'Cannot create index template {0}, {1}'.format(name, output)
         else:
             if check_definition:
-                definition_to_diff = {'aliases': {}, 'mappings': {}, 'settings': {}}
-                definition_to_diff.update(definition)
+                if type(definition) == str:
```
Comment (rallytime) [inline salt/states/elasticsearch.py]: I think using an `isinstance` check here is more clear.

File: salt/states/elasticsearch.py [hunk 2]
```diff
@@ -275,10 +275,16 @@ def index_template_present(name, definition, check_definition=False):
...
+                    definition_parsed = salt.utils.json.loads(definition)
+                else:
+                    definition_parsed = definition
                 current_template = __salt__['elasticsearch.index_template_get'](name=name)[name]
-                diff = __utils__['dictdiffer.deep_diff'](current_template, definition_to_diff)
+                # Prune empty keys (avoid false positive diff)
+                for key in ("mappings", "aliases", "settings"):
+                    if current_template[key] == {}:
```
Comment (rallytime) [inline salt/states/elasticsearch.py]: Same comment here. :)

**Review by rallytime — CHANGES_REQUESTED** [review]
Thanks for this PR @babs 

I have a couple of suggestions. In additon there is a very small lint error that needs to be fixed:

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/23133/violations/file/salt/states/elasticsearch.py/

File: salt/states/elasticsearch.py (hunk 2 above)
Comment (babs) [reply salt/states/elasticsearch.py]: The check here it not th know if it's a dict but and empty dict.
I find ```isinstance(current_template[key], dict) and len(current_template[key].key() == 0)``` a little bit too wordy for that purpose, any suggestion ?

File: salt/states/elasticsearch.py (hunk 1 above)
Comment (babs) [reply salt/states/elasticsearch.py]: I'll take care of that

**Comment by babs** [comment]
Hi @rallytime, should I squash the commits ?

File: salt/states/elasticsearch.py (hunk 2 above)
Comment (rallytime) [reply salt/states/elasticsearch.py]: What about `if not current_template[key]:`?

**Comment by cachedout** [comment]
Have we tested this change on EL 6 and above as well?

**Comment by babs** [comment]
@cachedout yes I did, I've both in prod using it (5.x and 6+).
The definition is slightly different at the state level but this is due to ES representation of data and a type change.

File: salt/states/elasticsearch.py (hunk 2 above)
Comment (babs) [reply salt/states/elasticsearch.py]: ```if not current_template[key]:``` would also match non dict typed values, I'll check if it might have any border effect as if an empty string was an expected value

Comment (rallytime) [reply salt/states/elasticsearch.py]: You're right and this is fine. It just felt that an awkward check to me.

**Review by rallytime — APPROVED** [review]
